icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -692,6 +772,7 @@ const MeasureUnitImpl& MeasureUnitImpl::forMeasureUnit(
     if (measureUnit.fImpl) {
         return *measureUnit.fImpl;
     } else {
+        // FIXME test this code path.
```

Comment:
By this FIXME, I'm reminding myself:

- the parser doesn't accept "" at the moment,
- getIdentifier() can return "" though, for MeasureUnit's not constructed via the parser,
- I'm expecting there to be a code path that fails therefore, of MeasureUnitImpl::forMeasureUnit can be invoked on a default-constructed MeasureUnit();

I need to convince myself that this can't happen (and then document it well enough to convince others), or I need to fix it.

Reply:
Done: my conclusion was the parser needs to support "" - it's used internally by factory functions (they parse the output of getIdentifier).

Marking as resolved.